### PR TITLE
Zjr/site changeset

### PR DESCRIPTION
**Pull request by zach-bob**
Added a URL validation step in the site changeset and added tests for the site changeset

### CODE REVIEW

File: lib/situation_room/site.ex [hunk 1]
```diff
@@ -7,10 +7,31 @@ defmodule SituationRoom.Site do
...
+          case :inet.gethostbyname(Kernel.to_charlist host) do
+            {:ok, _} -> nil
+            {:error, _} -> "invalid host"
+          end
+      end
+      |> case do
+        error when is_binary(error) -> [{field, Keyword.get(opts, :message, error)}]
+        _ -> []
+      end
+    end
```
Comment (cpjolicoeur) [inline lib/situation_room/site.ex:34]: Overall, I like this as a first pass, but I think we can probably improve on this a bit; both functionally and also idiomatically from an Elixir perspective.

First, we kind of doing two things here: 1) validating the format of the URI passed in (via using `URI.parse`) and 2) validating that the URI provided actually exists and is reachable (via `:inet.gethostbyname`)

I'm wondering if we should split these into two different validations?  Not sure on that yet, but looking for your thoughts.

For idiomatic elixir, we seem to be directly piping one case statement into another and it's a bit hard to follow.  I think we can better leverage a few languages features like pattern matching, or with clauses and things.  We can sit together and work through some of that if you'd prefer.

**Review by cpjolicoeur — CHANGES_REQUESTED** [review]

File: test/api/v1/validator_test.exs [hunk 2]
```diff
@@ -0,0 +1,49 @@
+defmodule API.ValidatorTest do
+  use ExUnit.Case
+
+  @test_suites [
+    [
+      {:n, "MojoTech Main"},
```
[the remark this answers is not in the document]
Comment (Zach-Bob) [reply test/api/v1/validator_test.exs:6]: That was the first way I found that I could iterate over and index them. I was using the atoms as keys. I will into refactoring to reduce complexity.
